larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -199,6 +170,35 @@ TfLiteStatus Prepare(KernelType kernel_type,
   TF_LITE_ENSURE_EQ(context, NumDimensions(post_activation_multiplier), 1);
   TF_LITE_ENSURE_EQ(context, NumDimensions(post_activation_bias), 1);
 
+  conv_params->read_bitpacked_input = input->type == kTfLiteInt32;
+  conv_params->write_bitpacked_output = output->type == kTfLiteInt32;
+
+  if (conv_params->padding_type == TfLitePadding::kTfLitePaddingSame &&
+      conv_params->pad_value != 1 && conv_params->write_bitpacked_output) {
+    context->ReportError(context,
+                         "Writing bitpacked output is only supported with "
+                         "valid or one-padding.");
+    return kTfLiteError;
+  }
+
+  // Read the input dimensions. TF Lite has the same input format as TensorFlow:
+  // (B, H, W, Ci).
+  conv_params->batch = input->dims->data[0];
+  conv_params->input_height = input->dims->data[1];
+  conv_params->input_width = input->dims->data[2];
+  if (!conv_params->read_bitpacked_input) {
+    TF_LITE_ENSURE_EQ(context, conv_params->channels_in, input->dims->data[3]);
```

Comment:
I just ran a sanity check with the benchmarking binary and this check fails on a model that was converted with an old version of the converter, should we remove this check to stay backwards compatible?
`larq_compute_engine/tflite/kernels/bconv2d.cc:190 conv_params->channels_in != input->dims->data[3] (0 != 64)`

Reply:
I don't have a strong opinion about this